Lib/test/_test_multiprocessing.py

```diff
@@ -3824,6 +3824,44 @@ def test_shared_memory_SharedMemoryServer_ignores_sigint(self):
 
         smm.shutdown()
 
+    def test_shared_memory_SharedMemoryManager_cleanup_if_parent_dies(self):
+        # If a process that spawned a SharedMemoryManager was terminated, its
+        # manager process should notice it, unlink the resources the manager
+        # delivered, and shut down.
+        cmd = '''if 1:
+            import sys, time
+
+            from multiprocessing.managers import SharedMemoryManager
+
+
+            smm = SharedMemoryManager()
+            smm.start()
+            sl = smm.ShareableList([1, 2, 3])
+
+            sys.stdout.write(f'{sl.shm.name}\\n')
+            sys.stdout.flush()
+            time.sleep(100)
+        '''
+
+        p = subprocess.Popen([sys.executable, '-E', '-c', cmd],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        shm_name = p.stdout.readline().decode().strip()
+        p.terminate()
+        start_time = time.monotonic()
+        deadline = start_time + 60
+        t = 0.1
+        while time.monotonic() < deadline:
+            time.sleep(t)
```

Comment:
I don't feel comfortable having tests that rely on sleep, they are almost assured to fail in some of the buildbots
 (check https://github.com/python/cpython/pull/10700 for example). Could you restructure this
tests to use synchronization primitives?

Reply:
`multiprocessing` remains the module with the higher amount of random failures due to sleeps/race conditions. I will dismiss my review in case there is no other way, but I still uneasy about adding more `sleep()` :(